lib/fastlane_core/configuration/config_item.rb

```diff
@@ -63,16 +66,37 @@ def valid?(value)
 
     # Returns an updated value type (if necessary)
     def auto_convert_value(value)
-      # Special treatment if the user specififed true, false or YES, NO
-      if %w(YES yes true).include?(value)
-        value = true
-      elsif %w(NO no false).include?(value)
-        value = false
+      # We must cast the type to String before comparing because of the way that Ruby
+      # handles class comparisons
+      case data_type.to_s
+      when 'Array'
+        value = value.split(',')
+      else
+        # Special treatment if the user specififed true, false or YES, NO
+        if %w(YES yes true).include?(value)
+          value = true
```

Comment:
How about we do this in a more flexible way, by moving the converting of those values into its own class? I'm more than happy to take this over, if you don't want to.